enclave_build/src/pull.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+
+use oci_distribution::{
+    client::{Client, ClientConfig, ClientProtocol, ImageData},
+    secrets::RegistryAuth,
+};
+
+use crate::{
+    image::{self, deserialize_from_reader},
+    EnclaveBuildError, Result,
+};
+
+/// Number of accepted image layer media types.
+pub const ACCEPTED_MEDIA_TYPES_COUNT: usize = 3;
+/// Accepted image layer media types.
+pub const ACCEPTED_MEDIA_TYPES: [&str; ACCEPTED_MEDIA_TYPES_COUNT] = [
+    oci_distribution::manifest::WASM_LAYER_MEDIA_TYPE,
+    oci_distribution::manifest::IMAGE_DOCKER_LAYER_GZIP_MEDIA_TYPE,
+    oci_distribution::manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE,
+];
+/// Sequence of environment variables and config file paths to try parsing: (env, path)
+pub const CONFIG_ENV_PATHS: [(&str, &str); 3] = [
+    ("DOCKER_CONFIG", "/"),
```

Comment:
Wouldn't that ending to `/` look for a directory rather than file itself?

Reply:
Yes, path string should be empty.